src/loader.js

```diff
@@ -362,7 +362,7 @@ function setupModuleLoader(window) {
           component: function(name, options) {
             function factory($injector) {
               function makeInjectable(fn) {
-                if (angular.isFunction(fn)) {
+                if (isFunction(fn) || isArray(fn)) {
```

Comment:
If that line didn't work every single service/directive/etc. you define would throw... (btw, it also has a bug of not handling array notation :smile:)